controller/scheduler/swarm.py

```diff
@@ -32,11 +30,15 @@ def create(self, name, image, command='', template=None, **kwargs):
             if mem[-2:-1].isalpha() and mem[-1].isalpha():
                 mem = mem[:-1]
         cpu = kwargs.get('cpu', {}).get(l['c_type'])
-        self.docker_cli.create_container(image=cimage, name=name,
-                                         command=command.encode('utf-8'), mem_limit=mem,
-                                         cpu_shares=cpu,
-                                         environment=[affinity],
-                                         host_config={'PublishAllPorts': True})
+        try:
+            self.docker_cli.create_container(image=cimage, name=name,
+                                             command=command.encode('utf-8'),
+                                             mem_limit=mem,
+                                             cpu_shares=cpu,
+                                             environment=[affinity],
+                                             host_config={'PublishAllPorts': True})
+        except errors.APIError as err:
+            raise errors.DockerException(str(err.message)+"  "+str(err.explanation))
```

Comment:
I'm confused why we're catching an `APIError` and re-raising it as a `DockerException`. They're both from docker-py's `errors` package, but I can't find any code that looks for those types specifically. How does this change any behavior?